File: Calculate_Pi_serial/calculate_pi_serial.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
  int n, i;
  double mypi, w, sum, x;

/* Take the number of intervals from the command line */
  if (argc!=2){
    printf("You must supply one command line argument, specifying the number of intervals\n");
    exit(1);
  }
  n = atoi(argv[1]);

/* Calculate the width of an interval */
  w = 1.0/(double)n;

  sum = 0.0;
  for(i=0; i<n; i++){
    x = ((double)i+0.5)*w;
    sum += 4.0/(1.0+x*x);
  }
  mypi = w*sum;
  printf("PI is approx %.16f\n", mypi);

  return 0;
}

File: Calculate_Pi_serial/calculate_pi_serial.f90
PROGRAM calculate_pi
  IMPLICIT NONE
  INTEGER :: n, i
  CHARACTER*80 :: input
  DOUBLE PRECISION :: mypi, w, sum, x

! Take the number of intervals from the command line
  IF (command_argument_count() /= 1) THEN
    print *, 'You must supply one command line argument, specifying the number of intervals'
    STOP
  END IF
  CALL get_command_argument(1,input)
  READ(input, FMT='(I20)') n

! Calculate the width of an interval
  w = 1.0D0/DBLE(n)

  sum = 0.0D0
  DO i=0, n-1
    x = (DBLE(i)+0.5D0)*w
    sum = sum + 4.0D0/(1.0D0+x*x)
  END DO
  mypi = w*sum

  print *, 'PI is approx ', mypi
END PROGRAM calculate_pi
